woocommerce/google-listings-and-ads
# API Pull - Notify when product changes

### Changes proposed in this Pull Request:

Closes part of #2146

This PR adds logic for calling the Notification Partner endpoint when a Product changes. 

In detail this PR:

- Add a Service NotificationsService. This service includes the topics, the API call to the partner notifications service in WPCOM, and a public method to notify() to perform the call.
- Tweak the hooks in SyncerHooks to call the notify() method 
- Logs when an error happens in the API Call 
- Tests for NotificationsService and tweak tests for SyncerHooks


### Detailed test instructions:


1. Checkout this PR
2. Be sure you are proxied in WPCOM and the branch you are is `trunk`
3. Authorize your APP and change the partner ID in the WPCOM `sites-woocommerce-partner-notifications.php` in the PARTNERS variable (this is because we don't have yet the final Partner APP)   
4. Create a product
5. Go to WooCommerce - Status - Logs  -- and see the error log
6. Update the product
7. Go to WooCommerce - Status - Logs  -- and see the error log
8. Delete the product 
9. Go to WooCommerce - Status - Logs  -- and see the error log

### Additional details:

- The Topics are not matching the topics in WPCOM Notification PArtners, this is due the new documentation for the endpoint and it's pending to update them in the WPCOM side. So in this PR we only need to test the call and also is a good chance to test the errors.
- I removed  `woocommerce_delete_product_variation` and `woocommerce_trash_product_variation` as they are duplicated with `trashed_post` and `deleted_post`
- After rebase PHPCS triggered some issues in `src/Infrastructure/GoogleListingsAndAdsPlugin.php` but they are not introduced here. So they are out Scope this PR



### Changelog entry

> Add - Notify when product changes
